### ISSUE TITLE
Don't list old metric names?

### ISSUE THREAD

**Issue by mzealey** [lomik/graphite-clickhouse#83]
Is there a way to filter the metric name list so that I can exclude metrics that have not been inserted in the past few days? I can see in the RU wiki page there is detail about the old index_tree table where you could set `Deleted = 1` but can't seem to find any documentation about how to do this currently?

**Comment by deniszh** [comment]
@mzealey : if your Clickhouse support deletion you can try same trick, but with DELETE
```
ALTER TABLE graphite_index DELETE 
WHERE (Path LIKE 'PREFIX.%')
GROUP BY Path
HAVING max(Version) < toUInt32(toDateTime(today() - 7))
```
But I'm not sure if DELETE WHERE supports GROUP BY ... HAVING clause.

**Comment by mzealey** [comment]
I imagine sometihng like this would work:

```
ALTER TABLE graphite_index
DELETE WHERE Path in (
  SELECT Path
  FROM graphite_index
  WHERE (Path LIKE 'PREFIX.%') GROUP BY Path HAVING max(Version) < toUInt32(toDateTime(today() - 7))
)
```

But there are also partial and reversed paths etc as well; do they not need deleting somehow?

**Comment by deniszh** [comment]
Yep, then SELECT should bit more complex. Reversed is easy, not sure about partial, though.

**Comment by Felixoid** [comment]
It looks related to https://github.com/lomik/graphite-clickhouse/issues/64

**Comment by bzed** [comment]
Also I think you want to select the Paths from graphite_index first, then remove the data from the graphite table and then tidy the graphite_index table? Does that make sense?

**Comment by spoofedpacket** [comment]
Could the 'delete' column features in recent clickhouse versions be leveraged for this? 
https://altinity.com/blog/clickhouse-replacingmergetree-explained-the-good-the-bad-and-the-ugly

We are using the ALTER TABLE method described above for some time now, while it works as expected it puts significant load on clickhouse so it would be great to have a lightweight alternative.

**Comment by Felixoid** [comment]
https://clickhouse.com/docs/en/sql-reference/statements/delete exists for quite a long.

**Comment by mzealey** [comment]
Yeah I'd use the lightweight deletes as above. Probably can just remove the 'alter table x delete' bit to 'delete from x'.
